wolfssl/test.h

```diff
@@ -182,7 +182,9 @@
     #include <sys/ioctl.h>
     #include <sys/time.h>
     #include <sys/socket.h>
-    #include <pthread.h>
+    #ifdef HAVE_PTHREAD
```

Comment:
Some Jenkins jobs are failing with this. The `HAVE_PTHREAD` comes from `config.h`. Please add this at top of test.h:

```
#ifdef HAVE_CONFIG_H
    #include <config.h>
#endif
```

Reply:
@dgarske we were talking about two different `test.h` files, one in [wolfssl/test.h](https://github.com/wolfSSL/wolfssl/blob/master/wolfssl/test.h) and the other in [wolfcrypt/test/test.h](https://github.com/wolfSSL/wolfssl/blob/master/wolfcrypt/test/test.h). 

I applied your suggested change to both, but the PRB is still failing.

The issue is a bit more complex in that `test.h` [uses](https://github.com/wolfSSL/wolfssl/blob/master/wolfssl/test.h#L2204) `_POSIX_THREADS` not `HAVE_PTHREAD`. There are of course a variety of ways and places to handle that.

What do you think of adding something like this to the end of [wolfssl/wolfcrypt/settings.h](https://github.com/wolfSSL/wolfssl/blob/2f44ece4d840354aed4a391ec408d01b706bbe57/wolfssl/wolfcrypt/settings.h#L3099) ?

```
#ifdef _POSIX_THREADS
    #define HAVE_PTHREAD 
#endif
```

Other suggestions?